Validation sample ends in error because it looks for identified keys for a deidentified sample

```
[INFO]	2023-11-08T00:49:17.970Z	e5925e83-ac19-4cd2-a1c9-65591f8c47ed	Rename external subject and external sample columns
2023-11-08 00:49:17,970 - INFO     - lambda_code               - lambda_handler                           : LineNo. 147  - Rename external subject and external sample columns
[ERROR] KeyError: 'date_of_birth'
Traceback (most recent call last):
  File "/var/task/lambda_code.py", line 165, in lambda_handler
    sample_df[date_column] = sample_df[date_column].apply(
  File "/var/lang/lib/python3.11/site-packages/pandas-1.5.3-py3.11-linux-x86_64.egg/pandas/core/frame.py", line 3807, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/var/lang/lib/python3.11/site-packages/pandas-1.5.3-py3.11-linux-x86_64.egg/pandas/core/indexes/base.py", line 3804, in get_loc
    raise KeyError(key) from err
```

--- a/deploy/cttso-ica-to-pieriandx-cdk/lambdas/get_metadata_from_portal_and_defaults_and_launch_validation_workflow/lambda_code.py
+++ b/deploy/cttso-ica-to-pieriandx-cdk/lambdas/get_metadata_from_portal_and_defaults_and_launch_validation_workflow/lambda_code.py
@@ -160,12 +160,6 @@ def lambda_handler(event, context):
         axis="columns"
     )
 
-    # Convert times to utc time and strings
-    for date_column in ["date_received", "date_collected", "date_of_birth"]:
-        sample_df[date_column] = sample_df[date_column].apply(
-            lambda x: datetime_obj_to_utc_isoformat(handle_date(x))
-        )
-
     # Assert expected values exist
     logger.info("Check we have all of the expected information")
     for expected_column in EXPECTED_ATTRIBUTES:
@@ -193,6 +187,14 @@ def lambda_handler(event, context):
             columns={
                 "external_subject_id": "study_subject_identifier"
             }
+        )
+
+    # Convert times to utc time and strings
+    for date_column in ["date_received", "date_collected", "date_of_birth"]:
+        if date_column not in sample_df.columns.tolist():
+            continue
+        sample_df[date_column] = sample_df[date_column].apply(
+            lambda x: datetime_obj_to_utc_isoformat(handle_date(x))
         )
 
     # Launch batch lambda function

